stream-chat-android-core/src/main/java/io/getstream/chat/android/core/internal/fsm/FiniteStateMachine.kt

```diff
@@ -52,48 +53,30 @@ public class FiniteStateMachine<S : Any, E : Any>(
         }
     }
 
+    /**
+     * The current state as [StateFlow].
+     */
+    public val stateFlow: StateFlow<S> = _state
+
     /**
      * The current state.
      */
     public val state: S
         get() = runBlocking {
-            mutex.withLockIfNotLocked { _state }
+            mutex.withLockIfNotLocked { _state.value }
         }
```

Comment:
It seems like we can now delete this construction since `MutableStateFlow.value` is a `Thread-safe` as per Doc.

So we could replace it with:
```kotlin
    public val state: S
        get() = _state.value
```

Reply:
Let me review it and update